tests/api.c

```diff
@@ -17799,6 +17945,163 @@ static void test_wolfSSL_X509_get_serialNumber(void)
+    AssertIntEQ(p_len, 35);
+    BIO_read(bio, (void*)rbuf, 35);
+    AssertStrEQ((char*)rbuf, "#040F48656C6C6F20776F6C6653534C2100");
+
+    /* Dump All + Show type */
+    XMEMSET(rbuf, 0, 255);
+    flags = ASN1_STRFLGS_DUMP_ALL | ASN1_STRFLGS_SHOW_TYPE;
+    p_len = wolfSSL_ASN1_STRING_print_ex(bio, asn_str, flags);
+    AssertIntEQ(p_len, 44);
+    BIO_read(bio, (void*)rbuf, 44);
+    AssertStrEQ((char*)rbuf, "OCTET STRING:#48656C6C6F20776F6C6653534C2100");
+
+    BIO_free(bio);
+    ASN1_STRING_free(asn_str);
+    ASN1_STRING_free(esc_str);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_ASN1_TIME_to_generalizedtime(void){
+#if defined(OPENSSL_EXTRA) && !defined(NO_ASN1_TIME)
+    WOLFSSL_ASN1_TIME *t;
+    WOLFSSL_ASN1_TIME *out;
+    WOLFSSL_ASN1_TIME *gtime;
+
+    printf(testingFmt, "wolfSSL_ASN1_TIME_to_generalizedtime()");
+
+    /* UTC Time test */
+    t = (WOLFSSL_ASN1_TIME*)XMALLOC(sizeof(WOLFSSL_ASN1_TIME), NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    XMEMSET(t->data, 0, ASN_GENERALIZED_TIME_SIZE);
+    out = (WOLFSSL_ASN1_TIME*)XMALLOC(sizeof(WOLFSSL_ASN1_TIME), NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    t->data[0] = ASN_UTC_TIME;
+    t->data[1] = ASN_UTC_TIME_SIZE;
+    XMEMCPY(t->data + 2,"050727123456Z",ASN_UTC_TIME_SIZE);
+
+    gtime = wolfSSL_ASN1_TIME_to_generalizedtime(t, &out);
+    AssertIntEQ(gtime->data[0], ASN_GENERALIZED_TIME);
+    AssertIntEQ(gtime->data[1], ASN_GENERALIZED_TIME_SIZE);
+    AssertStrEQ((char*)gtime->data + 2, "20050727123456Z");
+
+    /* Generalized Time test */
+    XMEMSET(t, 0, ASN_GENERALIZED_TIME_SIZE);
+    XMEMSET(out, 0, ASN_GENERALIZED_TIME_SIZE);
+    gtime = NULL;
+    t->data[0] = ASN_GENERALIZED_TIME;
+    t->data[1] = ASN_GENERALIZED_TIME_SIZE;
+    XMEMCPY(t->data + 2,"20050727123456Z",ASN_GENERALIZED_TIME_SIZE);
+    gtime = wolfSSL_ASN1_TIME_to_generalizedtime(t, &out);
+    AssertIntEQ(gtime->data[0], ASN_GENERALIZED_TIME);
+    AssertIntEQ(gtime->data[1], ASN_GENERALIZED_TIME_SIZE);
+    AssertStrEQ((char*)gtime->data + 2, "20050727123456Z");
+    XFREE(out, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+
+    /* Null parameter test */
+    XMEMSET(t, 0, ASN_GENERALIZED_TIME_SIZE);
+    gtime = NULL;
+    out = NULL;
+    t->data[0] = ASN_UTC_TIME;
+    t->data[1] = ASN_UTC_TIME_SIZE;
+    XMEMCPY(t->data + 2,"050727123456Z",ASN_UTC_TIME_SIZE);
+    AssertNotNull(gtime = wolfSSL_ASN1_TIME_to_generalizedtime(t, NULL));
+    AssertIntEQ(gtime->data[0], ASN_GENERALIZED_TIME);
+    AssertIntEQ(gtime->data[1], ASN_GENERALIZED_TIME_SIZE);
+    AssertStrEQ((char*)gtime->data + 2, "20050727123456Z");
+
+    XFREE(gtime, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    XFREE(t, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_X509_check_ca(void){
+#if defined(OPENSSL_EXTRA)
+    WOLFSSL_X509 *x509;
+
+    x509 = wolfSSL_X509_load_certificate_file(svrCertFile, WOLFSSL_FILETYPE_PEM);
+    AssertIntEQ(wolfSSL_X509_check_ca(x509), 1);
```

Comment:
Jenkins is failing both the "Disable Options Test" and "Testing known customer configurations" on the following configure commands:

**Disable Options Test**
./configure C_EXTRA_FLAGS="-fdebug-types-section -g1" --enable-opensslextra --enable-jobserver=4 --disable-rsa

**Testing known customer configurations:**
./configure C_EXTRA_FLAGS="-fdebug-types-section -g1" --enable-opensslextra --enable-jobserver=4 --enable-ecc --disable-rsa --enable-tls13

These configure commands both disable RSA, and are failing for the same reason.  When RSA is disabled, wolfSSL_X509_load_certificate_file() returns a NULL pointer, as it is not able to process the RSA-based certificate "svrCertFile".  "x509" should be checked against NULL and properly handled in the case where NO_RSA is defined.

Reply:
I added !defined(NO_RSA) directive.It looks good.
Thanks !